### PR TITLE
bug squashed! 👟🐛

### PR DESCRIPTION
**Pull request by nsafai**
Previously, any user who logged in was persisted on the app level. This PR fixes that by using session instead.

### CODE REVIEW

File: models/user.js [hunk 1]
```diff
@@ -8,43 +9,61 @@ const bcrypt = require('bcryptjs');
 const uniqueValidator = require('mongoose-unique-validator');
 
 const UserSchema = mongoose.Schema({
-  createdAt              : { type: Date  },
-  updatedAt              : { type: Date  },
+  createdAt              : { type: Date, default: Date.now },
```
Comment (PhyllisWong) [inline models/user.js]: new Date() vs Date.now: functionally similar except Date.now is not supported by Internet Explorer 8.

File: models/user.js [hunk 2]
```diff
@@ -8,43 +9,61 @@ const bcrypt = require('bcryptjs');
...
+    .then((user) => {
+      if (!user) {
+        const error = new Error('User not found.');
+        error.status = 401;
+        return next(error);
+      }
+      if (user) {
+        // User found, compare password
+        console.log(user);
+        bcrypt.compare(password, user.password, (error, isMatch) => {
```
Comment (PhyllisWong) [inline models/user.js]: where does isMatch come from?

File: models/user.js (hunk 1 above)
Comment (nsafai) [reply models/user.js]: that's good to know!

File: models/user.js (hunk 2 above)
Comment (nsafai) [reply models/user.js]: its "taco", can be called anything. It is the result of bcrypt.compare() and is either true or false.

Comment (PhyllisWong) [reply models/user.js]: 👍

**Pull request merged**